include/rocksdb/comparator.h

```diff
@@ -33,6 +33,35 @@ class CompareInterface {
   virtual int Compare(const Slice& a, const Slice& b) const = 0;
 };
 
+// The general interface for comparing two Slices that are related to
+// timestamps.
+// They are defined for both Comparator and some internal data structures.
+class CompareWithTimestampInterface {
+ public:
+  virtual ~CompareWithTimestampInterface() {}
+  int CompareWithoutTimestamp(const Slice& a, const Slice& b) const {
```

Comment:
Should this function be virtual?

Reply:
It is not virtual in the first place. I don't think we should create virtual functions when nobody use them.